game/hud/src/widgets/Crafting/components/Ingredients/index.tsx

```diff
@@ -0,0 +1,151 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * @Author: Mehuge (mehuge@sorcerer.co.uk)
+ * @Date: 2017-05-06 16:09:59
+ * @Last Modified by: Mehuge (mehuge@sorcerer.co.uk)
+ * @Last Modified time: 2017-06-15 21:48:51
+ */
+
+import * as React from 'react';
+import {connect} from 'react-redux';
+import { GlobalState } from '../../services/session/reducer';
+import { Ingredient } from '../../services/types';
+import IngredientItem from '../IngredientItem';
+import RepairItem from '../RepairItem';
+import PossibleIngredients from '../PossibleIngredients';
+import Select from '../Select';
+import Input from '../Input';
+import Label from '../Label';
+import Button from '../Button';
+import { StyleSheet, css, merge, ingredients as ingredientsStyles, IngredientsStyles} from '../../styles';
+
+import { InventoryItem } from '../../services/types';
+
+export interface IngredientsPropsRedux {
+  dispatch?: any;
+  ingredients?: Ingredient[];
+  havePossibleIngredients?: boolean;
+  status?: string;
+  job?: string;
+  loading?: boolean;
+}
+
+export interface IngredientsProps extends IngredientsPropsRedux{
+  add: (ingredient: Ingredient, qty: number) => void;
+  remove: (ingredient: Ingredient) => void;
+  style?: Partial<IngredientsStyles>;
+}
+
+const select = (state: GlobalState, props: IngredientsProps): IngredientsPropsRedux => {
```

Comment:
```
src/widgets/Crafting/components/Ingredients/index.tsx(151,32): error TS2345: Argument of type 'typeof Ingredients' is not assignable to parameter of type 'Component<IngredientsProps & DispatchProp<any> & IngredientsPropsRedux>'.
  Type 'typeof Ingredients' is not assignable to type 'StatelessComponent<IngredientsProps & DispatchProp<any> & IngredientsPropsRedux>'.
    Type 'typeof Ingredients' provides no match for the signature '(props: IngredientsProps & DispatchProp<any> & IngredientsPropsRedux & { children?: ReactNode; }, context?: any): ReactElement<any>'.
```

Referring to errors like this. Not sure exactly what they mean, was looking into them for a little but seems like @types/react updated and started throwing these errors all around our project. I'm not necessarily asking for a change, but an investigation as to what these errors mean. I pulled the error from the most recent TravisCI build and I was getting the error when I pulled down and did a clean `npm install` and build.